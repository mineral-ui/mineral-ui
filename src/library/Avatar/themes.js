/* @flow */
import type { AvatarThemeFn } from './types';

export const avatarTheme: AvatarThemeFn = (baseTheme) => ({
  Avatar_fontSize_small: baseTheme.fontSize_mouse,
  Avatar_fontSize_medium: baseTheme.fontSize_ui,
  Avatar_fontSize_large: baseTheme.h4_fontSize,
  Avatar_fontSize_jumbo: baseTheme.h4_fontSize,
  Avatar_fontWeight: baseTheme.fontWeight_bold,
  Avatar_size_small: baseTheme.size_small,
  Avatar_size_medium: baseTheme.size_medium,
  Avatar_size_large: baseTheme.size_large,
  Avatar_size_jumbo: baseTheme.size_jumbo,

  ...baseTheme
});
