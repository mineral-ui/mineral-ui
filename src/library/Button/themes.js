/* @flow */
import { pxToEm } from '../styles';

import type { ButtonThemeFn } from './types';

// prettier-ignore
export const buttonTheme: ButtonThemeFn = (baseTheme) => ({
  Button_backgroundColor: baseTheme.backgroundColor,
  Button_backgroundColor_active: baseTheme.backgroundColor_active,
  Button_backgroundColor_focus: baseTheme.backgroundColor_focus,
  Button_backgroundColor_hover: baseTheme.backgroundColor_hover,
  Button_backgroundColor_minimal_active: baseTheme.backgroundColor_active,
  Button_backgroundColor_minimal_hover: baseTheme.backgroundColor_hover,
  Button_backgroundColor_primary: baseTheme.backgroundColor_themePrimary,
  Button_backgroundColor_primary_active: baseTheme.backgroundColor_themePrimary_active,
  Button_backgroundColor_primary_focus: baseTheme.backgroundColor_themePrimary_focus,
  Button_backgroundColor_primary_hover: baseTheme.backgroundColor_themePrimary_hover,
  Button_borderColor: baseTheme.borderColor,
  Button_borderColor_active: baseTheme.borderColor_theme_active,
  Button_borderColor_focus: baseTheme.borderColor_theme_focus,
  Button_borderColor_hover: baseTheme.borderColor_theme_hover,
  Button_borderRadius: baseTheme.borderRadius_3,
  Button_borderWidth: 1, // px,
  Button_boxShadow_focus: `0 0 0 1px ${baseTheme.boxShadow_focusInner}, 0 0 0 2px ${baseTheme.borderColor_theme_focus}`,
  Button_color: baseTheme.color_theme,
  Button_color_minimal: baseTheme.color_theme,
  Button_color_primary: baseTheme.color_themePrimary,
  Button_fontWeight: baseTheme.fontWeight_semiBold,
  Button_paddingHorizontal: baseTheme.space_inset_sm_plus,
  Button_paddingIconOnly_small: pxToEm(3),
  Button_paddingIconOnly_medium: pxToEm(7),
  Button_paddingIconOnly_large: pxToEm(7),
  Button_paddingIconOnly_jumbo: pxToEm(13),
  Button_height_small: baseTheme.size_small,
  Button_height_medium: baseTheme.button_size_medium,
  Button_height_large: baseTheme.button_size_large,
  Button_height_jumbo: baseTheme.button_size_jumbo,

  ButtonContent_fontSize: baseTheme.fontSize_ui_sm,
  ButtonContent_fontSize_small: pxToEm(12),

  ButtonIcon_color: baseTheme.icon_color_theme,
  ButtonIcon_margin: baseTheme.space_inset_sm,

  ...baseTheme
});
