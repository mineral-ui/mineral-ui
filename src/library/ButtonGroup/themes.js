/* @flow */
import type { ButtonGroupThemeFn } from './types';

export const buttonGroupTheme: ButtonGroupThemeFn = (baseTheme) => ({
  ButtonGroupButton_backgroundColor_checkedDisabled: baseTheme.color_gray_40,
  ButtonGroupButton_border_disabled: `solid 1px ${baseTheme.borderColor}`,
  ButtonGroupButton_borderColor_active: baseTheme.borderColor_theme_active,
  ButtonGroupButton_borderColor_hover: baseTheme.borderColor_theme_hover,
  ButtonGroupButton_borderStartColor: baseTheme.borderColor,
  ButtonGroupButton_borderStartColor_checked: 'currentcolor',
  ButtonGroupButton_color_checkedDisabled: baseTheme.color_gray_60,

  ...baseTheme
});
