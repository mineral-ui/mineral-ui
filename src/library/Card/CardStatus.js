/* @flow */
import React from 'react';
import { createStyledComponent, pxToRem } from '../styles';
import IconDangerSimple from '../Icon/IconDangerSimple';
import IconSuccessSimple from '../Icon/IconSuccessSimple';
import IconWarningSimple from '../Icon/IconWarningSimple';
import CardRow from './CardRow';

type Props = {
  /** Status text */
  children: string,
  /** Available variants */
  variant: 'danger' | 'success' | 'warning'
};

export const componentTheme = (baseTheme: Object) => ({
  CardStatus_fontSize: baseTheme.fontSize_ui,
  CardStatus_fontWeight: baseTheme.fontWeight_regular,

  CardStatusIcon_margin: baseTheme.space_inline_sm,
  CardStatusIcon_size: pxToRem(12, baseTheme),

  ...baseTheme
});

const Root = createStyledComponent(
  CardRow,
  ({ theme: baseTheme, variant }) => {
    const theme = componentTheme(baseTheme);

    return {
      alignItems: 'center',
      color: theme[`color_${variant}`],
      display: 'flex',
      fontSize: theme.CardStatus_fontSize,
      fontWeight: theme.CardStatus_fontWeight,

      '& > [role="img"]': {
        height: theme.CardStatusIcon_size,
        [`margin${theme.rtlEnd}`]: theme.CardStatusIcon_margin,
        width: theme.CardStatusIcon_size
      }
    };
  },
  {
    displayName: 'CardStatus'
  }
);

const statusIcons = {
  danger: <IconDangerSimple />,
  success: <IconSuccessSimple />,
  warning: <IconWarningSimple />
};

/**
 * CardStatus provides a standard way of displaying a [Card's](/components/card) current status.
 */
export default function CardStatus(props: Props) {
  const { children, variant, ...restProps } = props;
  const rootProps = {
    variant,
    ...restProps
  };
  const icon = statusIcons[variant];

  return (
    <Root {...rootProps}>
      {icon}
      {children}
    </Root>
  );
}
