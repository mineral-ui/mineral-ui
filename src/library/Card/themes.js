/* @flow */
import { pxToEm } from '../styles';

import type {
  CardThemeFn,
  CardActionsThemeFn,
  CardBlockThemeFn,
  CardDividerThemeFn,
  CardFooterThemeFn,
  CardStatusThemeFn,
  CardTitleThemeFn
} from './types';

// prettier-ignore
export const cardTheme: CardThemeFn = (baseTheme) => ({
  Card_backgroundColor: baseTheme.panel_backgroundColor,
  Card_borderColor: baseTheme.panel_borderColor,
  Card_borderRadius: baseTheme.borderRadius_1,
  Card_boxShadow: baseTheme.boxShadow_2,
  Card_boxShadow_focus: `0 0 0 2px ${baseTheme.borderColor_theme_focus}, ${baseTheme.boxShadow_1}`,

  CardRow_marginVertical: baseTheme.space_stack_md,
  CardRow_marginVerticalLast: baseTheme.space_stack_lg,
  CardRow_paddingHorizontal: baseTheme.space_inset_md,

  ...baseTheme
});

export const cardActionsTheme: CardActionsThemeFn = (baseTheme) => ({
  CardActionsAction_spaceInline: baseTheme.space_inline_sm,

  ...baseTheme
});

export const cardBlockTheme: CardBlockThemeFn = (baseTheme) => ({
  CardBlock_fontSize: baseTheme.fontSize_ui,
  CardBlock_lineHeight: baseTheme.lineHeight_prose,

  ...baseTheme
});

export const cardDividerTheme: CardDividerThemeFn = (baseTheme) => ({
  CardDivider_borderColor: baseTheme.borderColor,
  CardDivider_borderWidth: '1px',

  ...baseTheme
});

export const cardFooterTheme: CardFooterThemeFn = (baseTheme) => ({
  CardFooter_backgroundColor: baseTheme.well_backgroundColor,
  CardFooter_borderColor: baseTheme.borderColor,

  CardFooterRow_marginVertical: baseTheme.space_stack_sm,
  CardFooterRow_marginVerticalLast: baseTheme.space_stack_md,

  CardFooterTitle_color: baseTheme.h5_color,
  CardFooterTitle_fontSize: baseTheme.h5_fontSize,
  CardFooterTitle_fontWeight: baseTheme.h5_fontWeight,

  ...baseTheme
});

export const cardStatusTheme: CardStatusThemeFn = (baseTheme) => ({
  CardStatus_fontSize: baseTheme.fontSize_ui,
  CardStatus_fontWeight: baseTheme.fontWeight_regular,

  CardStatusIcon_margin: baseTheme.space_inline_sm,
  CardStatusIcon_size: pxToEm(12),

  ...baseTheme
});

export const cardTitleTheme: CardTitleThemeFn = (baseTheme) => ({
  CardTitle_color: baseTheme.h4_color,
  CardTitle_fontSize: baseTheme.h4_fontSize,
  CardTitle_fontWeight: baseTheme.h4_fontWeight,

  CardTitleAvatar_margin: baseTheme.space_inline_sm,
  CardTitleAvatarSize: baseTheme.size_small,
  CardTitleAvatarSize_large: baseTheme.size_large,

  CardTitleIcon_margin: baseTheme.space_inline_sm,

  CardTitleSecondaryText_color: baseTheme.color_mouse,
  CardTitleSecondaryText_fontSize: baseTheme.fontSize_mouse,
  CardTitleSecondaryText_fontWeight: baseTheme.fontWeight_regular,

  CardSubtitle_color: baseTheme.color_mouse,
  CardSubtitle_fontSize: baseTheme.fontSize_mouse,
  CardSubtitle_fontWeight: baseTheme.fontWeight_regular,
  CardSubtitle_marginTop: baseTheme.space_stack_sm,

  ...baseTheme
});
