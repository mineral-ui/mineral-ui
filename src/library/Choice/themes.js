/* @flow */
import { pxToEm } from '../styles';

import type { ChoiceThemeFn, ChoiceGroupThemeFn } from './types';

export const choiceTheme: ChoiceThemeFn = (baseTheme) => {
  const colors = {
    background: baseTheme.input_backgroundColor,
    regular: baseTheme.borderColor,
    checked: baseTheme.borderColor_theme,
    focus: baseTheme.borderColor_theme_focus,
    hover: baseTheme.borderColor_theme_hover
  };

  // prettier-ignore
  return {
    ChoiceControl_backgroundColor: colors.background,
    ChoiceControl_backgroundColor_checked: colors.checked,
    ChoiceControl_backgroundColor_checkedHover: colors.hover,
    ChoiceControl_borderColor: colors.regular,
    ChoiceControl_borderColor_hover: colors.hover,
    ChoiceControl_borderColor_checked: colors.checked,
    ChoiceControl_borderColor_checkedHover: colors.hover,
    ChoiceControl_borderRadius: baseTheme.borderRadius_1,
    ChoiceControl_boxShadow_focus: `0 0 0 1px ${baseTheme.boxShadow_focusInner}, 0 0 0 2px ${colors.focus}`,
    ChoiceControl_size: pxToEm(16),
    ChoiceControl_size_jumbo: pxToEm(24),

    ChoiceText_color: baseTheme.color,
    ChoiceText_fontSize: baseTheme.fontSize_ui,
    ChoiceText_fontSize_small: pxToEm(12),
    ChoiceText_marginHorizontal: baseTheme.space_inline_md,

    ...baseTheme
  };
};

export const choiceGroupTheme: ChoiceGroupThemeFn = (baseTheme) => ({
  ChoiceGroupControl_marginHorizontal_inline: baseTheme.space_inline_xl,
  ChoiceGroupControl_marginVertical_stacked: baseTheme.space_stack_md,
  ChoiceGroupControl_marginVertical_stackedJumbo: baseTheme.space_stack_lg,

  ...baseTheme
});
