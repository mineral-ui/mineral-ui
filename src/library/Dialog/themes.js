/* @flow */
import { pxToEm } from '../styles';
import { mapComponentThemes } from '../themes';
import { overflowContainerWithShadowsTheme } from '../OverflowContainer/themes';

import type {
  DialogThemeFn,
  DialogActionsThemeFn,
  DialogBodyThemeFn,
  DialogRowThemeFn,
  DialogTitleThemeFn
} from './types';

export const dialogTheme: DialogThemeFn = (baseTheme) => ({
  Dialog_transitionDuration: '250ms',
  Dialog_zIndex: baseTheme.zIndex_1600,

  DialogCloseButton_margin: baseTheme.space_inline_sm,

  DialogContent_backgroundColor: baseTheme.panel_backgroundColor,
  DialogContent_borderColor: baseTheme.panel_borderColor,
  DialogContent_borderRadius: baseTheme.borderRadius_1,
  DialogContent_boxShadow: baseTheme.boxShadow_5,
  DialogContent_maxHeight: '80vh',
  DialogContent_maxHeight_small: pxToEm(560),
  DialogContent_maxHeight_medium: pxToEm(560),
  DialogContent_maxHeight_large: pxToEm(720),
  DialogContent_maxWidth_small: pxToEm(400),
  DialogContent_maxWidth_medium: pxToEm(640),
  DialogContent_maxWidth_large: pxToEm(1200),
  DialogContent_minWidth: pxToEm(360),
  DialogContent_offsetVertical: baseTheme.space_stack_xxl,
  DialogContent_width_small: '35vw',
  DialogContent_width_medium: '50vw',
  DialogContent_width_large: '80vw',

  DialogOverlay_backgroundColor: 'rgba(0, 0, 0, 0.6)',

  ...dialogRowTheme(baseTheme),
  ...baseTheme
});

export const dialogActionsTheme: DialogActionsThemeFn = (baseTheme) => ({
  DialogActionsItem_margin: baseTheme.space_stack_sm,

  ...baseTheme
});

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const dialogBodyTheme: DialogBodyThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'OverflowContainerWithShadows',
      theme: overflowContainerWithShadowsTheme(baseTheme)
    },
    {
      name: 'DialogBody',
      theme: {}
    },
    baseTheme
  );

export const dialogRowTheme: DialogRowThemeFn = (baseTheme) => ({
  DialogRow_fontSize: baseTheme.fontSize_ui,
  DialogRow_paddingHorizontal: baseTheme.space_inline_lg,
  DialogRow_marginVertical: baseTheme.space_stack_lg,

  ...baseTheme
});

export const dialogTitleTheme: DialogTitleThemeFn = (baseTheme) => ({
  DialogTitle_color: baseTheme.h4_color,
  DialogTitle_fontSize: baseTheme.h4_fontSize,
  DialogTitle_fontWeight: baseTheme.h4_fontWeight,

  DialogTitleIcon_margin: baseTheme.space_inline_sm,
  DialogTitleIcon_size: pxToEm(24),

  ...baseTheme
});
