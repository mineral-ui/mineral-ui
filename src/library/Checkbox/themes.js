/* @flow */
import { mapComponentThemes } from '../themes';
import { choiceTheme, choiceGroupTheme } from '../Choice';

import type { CheckboxThemeFn, CheckboxGroupThemeFn } from './types';

import { pxToEm } from '../styles';

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const checkboxTheme: CheckboxThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'Choice',
      theme: choiceTheme(baseTheme)
    },
    {
      name: 'Checkbox',
      theme: {
        CheckboxControl_backgroundColor_checked: baseTheme.checkbox_backgroundColor_checked,
        CheckboxControl_backgroundColor_checkedHover: baseTheme.checkbox_backgroundColor_checkedHover,
        CheckboxControl_borderColor: baseTheme.checkbox_borderColor,
        CheckboxControl_borderColor_checked: baseTheme.checkbox_borderColor_checked,
        CheckboxControl_borderColor_checkedHover: baseTheme.checkbox_borderColor_checkedHover,
        CheckboxControl_borderColor_hover: baseTheme.checkbox_borderColor_hover,
        CheckboxControl_borderRadius: baseTheme.borderRadius_2,
        CheckboxControl_borderWidth: `${baseTheme.checkbox_borderWidth}px`,
        CheckboxControl_size: pxToEm(20),
        CheckboxText_marginHorizontal: baseTheme.space_inline_sm
      }
    },
    baseTheme
  );

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const checkboxGroupTheme: CheckboxGroupThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'ChoiceGroup',
      theme: choiceGroupTheme(baseTheme)
    },
    {
      name: 'CheckboxGroup',
      theme: {}
    },
    baseTheme
  );
