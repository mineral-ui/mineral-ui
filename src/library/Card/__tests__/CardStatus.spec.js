/* @flow */
import React from 'react';
import { shallow } from 'enzyme';
import CardStatus, { componentTheme } from '../../Card/CardStatus';
import testThemeOverrides from '../../../../utils/testThemeOverrides';
import { getProcessedComponentThemeKeys } from '../../themes/processComponentTheme';

function shallowCardStatus(props = {}) {
  const cardStatusProps = {
    children: 'children',
    variant: 'danger',
    ...props
  };
  return shallow(<CardStatus {...cardStatusProps} />);
}

describe('CardStatus', () => {
  it('renders', () => {
    const cardStatus = shallowCardStatus();

    expect(cardStatus.exists()).toEqual(true);
  });

  describe('theme overrides', () => {
    testThemeOverrides(
      <CardStatus variant="danger">children</CardStatus>,
      getProcessedComponentThemeKeys(componentTheme)
    );
  });
});
