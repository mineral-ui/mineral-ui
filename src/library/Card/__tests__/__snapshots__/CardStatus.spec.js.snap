// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardStatus demo examples Snapshots: basic 1`] = `
.emotion-7 {
  margin-bottom: 1em;
  margin-top: 1em;
  padding-left: 1em;
  padding-right: 1em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #de1b1b;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.875em;
  font-weight: 400;
}

.emotion-7 > [role="img"] {
  height: 0.75em;
  margin-right: 0.5em;
  width: 0.75em;
}

.emotion-6 {
  fill: currentcolor;
  font-size: 16px;
  height: 1em;
  width: 1em;
}

.emotion-27::after {
  clear: both;
  content: "";
  display: table;
}

.emotion-27 > * {
  margin-bottom: 1em;
}

@media (min-width:40em) {
  .emotion-27 > * {
    float: left;
    margin-right: 1em;
    width: 47%;
  }
}

@media (min-width:64em) {
  .emotion-27 > *:nth-last-child(n+3),
  .emotion-27 > *:nth-last-child(n+3) ~ * {
    width: 30%;
  }
}

.emotion-8 {
  box-sizing: border-box;
  color: #333840;
  font-family: "Open Sans",-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
  background-color: #ffffff;
  border: 1px solid #ebeff5;
  border-radius: 0.1875em;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2),0 4px 8px 0 rgba(0,0,0,0.2);
  padding-bottom: 0.01em;
  padding-top: 0.01em;
}

.emotion-8 *,
.emotion-8 *::before,
.emotion-8 *::after {
  box-sizing: inherit;
}

.emotion-8:focus {
  box-shadow: 0 0 0 2px #1d5bbf,0 1px 2px 0 rgba(0,0,0,0.2),0 2px 4px 0 rgba(0,0,0,0.2);
}

.emotion-3 {
  margin-bottom: 1em;
  margin-top: 1em;
  padding-left: 1em;
  padding-right: 1em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-1 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-1 > [role="img"] {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 0.5em;
  position: relative;
  top: 0.25em;
}

.emotion-0 {
  color: #58606e;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: 1.125em;
  font-weight: 700;
  margin: 0;
}

.emotion-5 {
  margin-bottom: 1em;
  margin-top: 1em;
  padding-left: 1em;
  padding-right: 1em;
}

.emotion-5:last-child {
  margin-bottom: 1.5em;
}

.emotion-4 {
  font-size: 0.875em;
  line-height: 1.5;
}

.emotion-16 {
  margin-bottom: 1em;
  margin-top: 1em;
  padding-left: 1em;
  padding-right: 1em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #2a854e;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.875em;
  font-weight: 400;
}

.emotion-16 > [role="img"] {
  height: 0.75em;
  margin-right: 0.5em;
  width: 0.75em;
}

.emotion-25 {
  margin-bottom: 1em;
  margin-top: 1em;
  padding-left: 1em;
  padding-right: 1em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #ad5f00;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.875em;
  font-weight: 400;
}

.emotion-25 > [role="img"] {
  height: 0.75em;
  margin-right: 0.5em;
  width: 0.75em;
}

<Styled(div)>
  <div
    className="emotion-27"
  >
    <Card>
      <Card>
        <div
          className="emotion-8"
        >
          <CardTitle>
            <CardTitle>
              <div
                className="emotion-3"
              >
                <Styled(div)>
                  <div
                    className="emotion-2"
                  >
                    <Styled(div)>
                      <div
                        className="emotion-1"
                      >
                        <Styled(h3)>
                          <h3
                            className="emotion-0"
                          >
                            Card Title
                          </h3>
                        </Styled(h3)>
                      </div>
                    </Styled(div)>
                  </div>
                </Styled(div)>
              </div>
            </CardTitle>
          </CardTitle>
          <CardBlock>
            <CardBlock>
              <div
                className="emotion-5"
              >
                <Styled(div)>
                  <div
                    className="emotion-4"
                  >
                    
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec mattis
  pretium massa. Aliquam erat volutpat. Nulla facilisi.

                  </div>
                </Styled(div)>
              </div>
            </CardBlock>
          </CardBlock>
          <CardStatus
            variant="danger"
          >
            <CardStatus
              variant="danger"
            >
              <div
                className="emotion-7"
              >
                <IconDangerSimple>
                  <Icon
                    rtl={false}
                    size="medium"
                  >
                    <Styled(svg)
                      aria-hidden={true}
                      focusable="false"
                      role="img"
                      rtl={false}
                      size="medium"
                      viewBox="0 0 24 24"
                    >
                      <svg
                        aria-hidden={true}
                        className="emotion-6"
                        focusable="false"
                        role="img"
                        viewBox="0 0 24 24"
                      >
                        <g>
                          <path
                            d="M3.94 20.49h16.118a1 1 0 0 0 .866-1.498l-8.06-13.99a.996.996 0 0 0-1.732-.001L3.074 18.993a.998.998 0 0 0 .867 1.499l-.001-.002z"
                          />
                        </g>
                      </svg>
                    </Styled(svg)>
                  </Icon>
                </IconDangerSimple>
                Unavailable
              </div>
            </CardStatus>
          </CardStatus>
        </div>
      </Card>
    </Card>
    <Card>
      <Card>
        <div
          className="emotion-8"
        >
          <CardTitle>
            <CardTitle>
              <div
                className="emotion-3"
              >
                <Styled(div)>
                  <div
                    className="emotion-2"
                  >
                    <Styled(div)>
                      <div
                        className="emotion-1"
                      >
                        <Styled(h3)>
                          <h3
                            className="emotion-0"
                          >
                            Card Title
                          </h3>
                        </Styled(h3)>
                      </div>
                    </Styled(div)>
                  </div>
                </Styled(div)>
              </div>
            </CardTitle>
          </CardTitle>
          <CardBlock>
            <CardBlock>
              <div
                className="emotion-5"
              >
                <Styled(div)>
                  <div
                    className="emotion-4"
                  >
                    
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec mattis
  pretium massa. Aliquam erat volutpat. Nulla facilisi.

                  </div>
                </Styled(div)>
              </div>
            </CardBlock>
          </CardBlock>
          <CardStatus
            variant="success"
          >
            <CardStatus
              variant="success"
            >
              <div
                className="emotion-16"
              >
                <IconSuccessSimple>
                  <Icon
                    rtl={false}
                    size="medium"
                  >
                    <Styled(svg)
                      aria-hidden={true}
                      focusable="false"
                      role="img"
                      rtl={false}
                      size="medium"
                      viewBox="0 0 24 24"
                    >
                      <svg
                        aria-hidden={true}
                        className="emotion-6"
                        focusable="false"
                        role="img"
                        viewBox="0 0 24 24"
                      >
                        <g>
                          <path
                            d="M12 3c4.968 0 9 4.032 9 9s-4.032 9-9 9-9-4.032-9-9 4.032-9 9-9z"
                          />
                        </g>
                      </svg>
                    </Styled(svg)>
                  </Icon>
                </IconSuccessSimple>
                Available
              </div>
            </CardStatus>
          </CardStatus>
        </div>
      </Card>
    </Card>
    <Card>
      <Card>
        <div
          className="emotion-8"
        >
          <CardTitle>
            <CardTitle>
              <div
                className="emotion-3"
              >
                <Styled(div)>
                  <div
                    className="emotion-2"
                  >
                    <Styled(div)>
                      <div
                        className="emotion-1"
                      >
                        <Styled(h3)>
                          <h3
                            className="emotion-0"
                          >
                            Card Title
                          </h3>
                        </Styled(h3)>
                      </div>
                    </Styled(div)>
                  </div>
                </Styled(div)>
              </div>
            </CardTitle>
          </CardTitle>
          <CardBlock>
            <CardBlock>
              <div
                className="emotion-5"
              >
                <Styled(div)>
                  <div
                    className="emotion-4"
                  >
                    
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec mattis
  pretium massa. Aliquam erat volutpat. Nulla facilisi.

                  </div>
                </Styled(div)>
              </div>
            </CardBlock>
          </CardBlock>
          <CardStatus
            variant="warning"
          >
            <CardStatus
              variant="warning"
            >
              <div
                className="emotion-25"
              >
                <IconWarningSimple>
                  <Icon
                    rtl={false}
                    size="medium"
                  >
                    <Styled(svg)
                      aria-hidden={true}
                      focusable="false"
                      role="img"
                      rtl={false}
                      size="medium"
                      viewBox="0 0 24 24"
                    >
                      <svg
                        aria-hidden={true}
                        className="emotion-6"
                        focusable="false"
                        role="img"
                        viewBox="0 0 24 24"
                      >
                        <g>
                          <path
                            d="M13.414 2.718l7.868 7.868c.78.78.78 2.047 0 2.828l-7.868 7.868c-.78.78-2.047.78-2.828 0l-7.868-7.868a2.001 2.001 0 0 1 0-2.828l7.868-7.868c.78-.78 2.047-.78 2.828 0z"
                          />
                        </g>
                      </svg>
                    </Styled(svg)>
                  </Icon>
                </IconWarningSimple>
                Warning
              </div>
            </CardStatus>
          </CardStatus>
        </div>
      </Card>
    </Card>
  </div>
</Styled(div)>
`;

exports[`CardStatus demo examples Snapshots: rtl 1`] = `
.emotion-2 {
  fill: currentcolor;
  font-size: 16px;
  height: 1em;
  width: 1em;
}

.emotion-5::after {
  clear: both;
  content: "";
  display: table;
}

.emotion-5 > * {
  margin-bottom: 1em;
}

@media (min-width:40em) {
  .emotion-5 > * {
    float: left;
    margin-right: 1em;
    width: 47%;
  }
}

@media (min-width:64em) {
  .emotion-5 > *:nth-last-child(n+3),
  .emotion-5 > *:nth-last-child(n+3) ~ * {
    width: 30%;
  }
}

.emotion-4 {
  box-sizing: border-box;
  color: #333840;
  font-family: "Open Sans",-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
  background-color: #ffffff;
  border: 1px solid #ebeff5;
  border-radius: 0.1875em;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2),0 4px 8px 0 rgba(0,0,0,0.2);
  padding-bottom: 0.01em;
  padding-top: 0.01em;
}

.emotion-4 *,
.emotion-4 *::before,
.emotion-4 *::after {
  box-sizing: inherit;
}

.emotion-4:focus {
  box-shadow: 0 0 0 2px #1d5bbf,0 1px 2px 0 rgba(0,0,0,0.2),0 2px 4px 0 rgba(0,0,0,0.2);
}

.emotion-1 {
  margin-bottom: 1em;
  margin-top: 1em;
  padding-left: 1em;
  padding-right: 1em;
}

.emotion-1:last-child {
  margin-bottom: 1.5em;
}

.emotion-0 {
  font-size: 0.875em;
  line-height: 1.5;
}

.emotion-3 {
  margin-bottom: 1em;
  margin-top: 1em;
  padding-left: 1em;
  padding-right: 1em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #de1b1b;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.875em;
  font-weight: 400;
}

.emotion-3 > [role="img"] {
  height: 0.75em;
  margin-left: 0.5em;
  width: 0.75em;
}

<Styled(div)
  dir="rtl"
>
  <div
    className="emotion-5"
    dir="rtl"
  >
    <ThemeProvider>
      <ThemeProvider>
        <Card>
          <Card>
            <div
              className="emotion-4"
            >
              <CardBlock>
                <CardBlock>
                  <div
                    className="emotion-1"
                  >
                    <Styled(div)>
                      <div
                        className="emotion-0"
                      >
                        
  خدمة الجزر أبجد هوز. حتى على مستوى أسعار العقارات. الغلوتين. لا عقوبة. حتى الرعاية. ولكن الآن تنورة المؤلف والتغذية.

                      </div>
                    </Styled(div)>
                  </div>
                </CardBlock>
              </CardBlock>
              <CardStatus
                variant="danger"
              >
                <CardStatus
                  variant="danger"
                >
                  <div
                    className="emotion-3"
                  >
                    <IconDangerSimple>
                      <Icon
                        rtl={false}
                        size="medium"
                      >
                        <Styled(svg)
                          aria-hidden={true}
                          focusable="false"
                          role="img"
                          rtl={false}
                          size="medium"
                          viewBox="0 0 24 24"
                        >
                          <svg
                            aria-hidden={true}
                            className="emotion-2"
                            focusable="false"
                            role="img"
                            viewBox="0 0 24 24"
                          >
                            <g>
                              <path
                                d="M3.94 20.49h16.118a1 1 0 0 0 .866-1.498l-8.06-13.99a.996.996 0 0 0-1.732-.001L3.074 18.993a.998.998 0 0 0 .867 1.499l-.001-.002z"
                              />
                            </g>
                          </svg>
                        </Styled(svg)>
                      </Icon>
                    </IconDangerSimple>
                    غير متوفره
                  </div>
                </CardStatus>
              </CardStatus>
            </div>
          </Card>
        </Card>
      </ThemeProvider>
    </ThemeProvider>
  </div>
</Styled(div)>
`;
